client/albion_watcher.go

```diff
@@ -0,0 +1,76 @@
+package client
+
+import (
+	"log"
+	"time"
+
+	"github.com/google/gopacket/pcap"
+)
+
+type albionProcessWatcher struct {
+	pid       int
+	known     []int
+	listeners map[int][]*listener
+	quit      chan bool
+}
+
+func newAlbionProcessWatcher(pid int) *albionProcessWatcher {
+	return &albionProcessWatcher{
+		pid:  pid,
+		listeners: make(map[int][]*listener),
+		quit: make(chan bool),
+	}
+}
+
+func (apw *albionProcessWatcher) run() {
+	log.Printf("Watching Albion process with PID \"%d\"...", apw.pid)
+	r := newRouter()
+
+	for {
+		select {
+		case <-apw.quit:
```

Comment:
There are a few reasons I like this better:

The other way, `for !PollChannel(quit) {` , doesn't read very well. For not poll channel "quit"? Anyone new coming along to the code isn't going to understand that until they go and understand the PollChannel function whereas this is a very standard way to handle it in Go.

This way also leaves us open to inserting more cases without a lot of refactor and it's consistent with other implementations where we need the switch case anyways.